### PR TITLE
Change the description of xml subtype

### PR DESCRIPTION
**Pull request by KavinduZoysa**
## Purpose
$subject

### CODE REVIEW

File: examples/xml-subtyping/xml_subtyping.description [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
-// An `xml` value belongs to `xml:Text` if it consists of a text item or is 
-// empty.
-// An `xml` value belongs to the type `xml<T>` if each of its members belong 
-// to `T`.
-// Functions in lang.xml use this to provide safe and convenient typing.
-// For example, `x.elements()` returns element items in `x` as type 
-// `xml<xml:Element>` and `e.getName()` and `e.setName()` are defined when 
-// `e` has type `xml:Element`.
+// <ul>
+// <li>An `xml` value belongs to `xml:Element` if it consists of just an element item.</li>
```
Comment (praneesha) [inline examples/xml-subtyping/xml_subtyping.description:2]:
```suggestion
// <li>An `xml` value belongs to the `xml:Element` if it consists of just one element item.</li>
```

File: examples/xml-subtyping/xml_subtyping.description [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
+// <li>Similarly for `xml:Comment` and `xml:ProcessingInstruction`.</li>
```
Comment (praneesha) [inline examples/xml-subtyping/xml_subtyping.description:3]:
```suggestion
// <li>It is the same for the `xml:Comment` and `xml:ProcessingInstruction`.</li>
```

File: examples/xml-subtyping/xml_subtyping.description [hunk 3]
```diff
@@ -1,10 +1,12 @@
...
+// <li>An `xml` value belongs to `xml:Text` if it consists of a text item or is empty.</li>
+// <li>An `xml` value belongs to the type `xml<T>` if each of its members belong to `T`.</li>
+// <li>Functions in lang.xml use this to provide safe and convenient typing. For example,
```
Comment (praneesha) [inline examples/xml-subtyping/xml_subtyping.description:6]:
```suggestion
// <li>Functions in the `lang.xml` use this to provide safe and convenient typing. For example, 
```

File: examples/xml-subtyping/xml_subtyping.description [hunk 4]
```diff
@@ -1,10 +1,12 @@
...
-// `e` has type `xml:Element`.
+// <ul>
+// <li>An `xml` value belongs to `xml:Element` if it consists of just an element item.</li>
+// <li>Similarly for `xml:Comment` and `xml:ProcessingInstruction`.</li>
+// <li>An `xml` value belongs to `xml:Text` if it consists of a text item or is empty.</li>
+// <li>An `xml` value belongs to the type `xml<T>` if each of its members belong to `T`.</li>
+// <li>Functions in lang.xml use this to provide safe and convenient typing. For example, 
+// <ul>
+// <li>`x.elements()` returns element items in `x` as type `xml<xml:Element>.`</li>
+// <li>`e.getName()` and `e.setName()` are defined when `e` has type `xml:Element`.</li>
```
Comment (praneesha) [inline examples/xml-subtyping/xml_subtyping.description:9]:
```suggestion
// <li>`e.getName()` and `e.setName()` are defined when `e` has the type `xml:Element`.</li>
```

Comment (KavinduZoysa) [reply examples/xml-subtyping/xml_subtyping.description]: Fixed

File: examples/xml-subtyping/xml_subtyping.description (hunk 3 above)
Comment (KavinduZoysa) [reply examples/xml-subtyping/xml_subtyping.description]: FIxed

File: examples/xml-subtyping/xml_subtyping.description (hunk 2 above)
Comment (KavinduZoysa) [reply examples/xml-subtyping/xml_subtyping.description]: Fixed

File: examples/xml-subtyping/xml_subtyping.description (hunk 1 above)
Comment (KavinduZoysa) [reply examples/xml-subtyping/xml_subtyping.description]: Fixed

**Review by praneesha — APPROVED** [review]

File: examples/xml-subtyping/xml_subtyping.description [hunk 5]
```diff
@@ -1,10 +1,12 @@
...
+// <li>It is the same for the `xml:Comment` and `xml:ProcessingInstruction`.</li>
+// <li>An `xml` value belongs to `xml:Text` if it consists of a text item or is empty.</li>
+// <li>An `xml` value belongs to the type `xml<T>` if each of its members belong to `T`.</li>
+// <li>Functions in the `lang.xml` use this to provide safe and convenient typing. For example,
+// <ul>
+// <li>`x.elements()` returns element items in `x` as type `xml<xml:Element>.`</li>
+// <li>`e.getName()` and `e.setName()` are defined when `e` has the type `xml:Element`.</li>
+// </ul>
+// </li>
+// </ul>
```
Comment (pubudu91) [inline examples/xml-subtyping/xml_subtyping.description:12]: Er why change these paras to lists? 🤔

Comment (KavinduZoysa) [reply examples/xml-subtyping/xml_subtyping.description:12]: Having them as point form is easy to read. That's why I did this change. If not let's ignore this change

Comment (MaryamZi) [reply examples/xml-subtyping/xml_subtyping.description:12]: I added a comment for https://github.com/ballerina-platform/ballerina-distribution/pull/2550#discussion_r795043364 also. IMO, the entire description being a list may not look nice.

Anyway these are repeated in-line also right? Maybe eventually we can replace the current text with a better overview, but I don't think we should be making the whole thing a list now.

Comment (KavinduZoysa) [reply examples/xml-subtyping/xml_subtyping.description:12]: Let's close this. It is good to keep it in the existing way

**Pull request closed without merging**